packages/gestalt/src/Masonry.tsx

```diff
@@ -595,6 +655,39 @@ export default class Masonry<T> extends ReactComponent<Props<T>, State<T>> {
       // When the width is empty (usually after a re-mount) render an empty
       // div to collect the width for layout
       gridBody = <div ref={this.setGridWrapperRef} style={{ width: '100%' }} />;
+    } else if (renderLoadingState) {
+      getPositions = defaultLayout({
+        align,
+        measurementCache: measurementStore,
+        positionCache: positionStore,
+        columnWidth,
+        gutter,
+        layout,
+        minCols,
+        rawItemCount: _loadingStateItems.length,
+        width,
+        logWhitespace: _logTwoColWhitespace,
+        _getColumnSpanConfig,
+        renderLoadingState,
+      });
+      const positions = getPositions(_loadingStateItems as any);
```

Comment:
One option could be to have two props on the getPositions function, and that the second prop is for loading items ({height: ...}), that way we can clearly separate both cases and usa that inside the layout functions to get the height from the cache in the normal items case but use th height prop on the loading items case.

Reply:
if we update the function signature for one, we should do so for all the layout functions. while the inputs are different, the current expectation is that the return function signature is the same (which we shouldn't break). 

alternatively, we could also make the input type polymorphic `(items: ReaonlyArray<T> | ReadonlyArray<LoadingStateItem>)`

Reply:
addressed in https://github.com/pinterest/gestalt/pull/3637#discussion_r1681456191